js_modules/dagster-ui/packages/ui-core/src/asset-graph/useLiveDataForAssetKeysBatcher.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+import {AssetKeyInput} from '../graphql/types';
+
+import {buildLiveData} from './Utils';
+export const useLiveDataForAssetKeysBatcher = (assetKeys: AssetKeyInput[], batchSize = 1) => {
+  const [_, forceRerender] = React.useReducer((s) => s + 1, 0);
+  const currentBatchIndexRef = React.useRef(0);
+  const currentBatchIndex = currentBatchIndexRef.current;
+
+  function setBatchIndex(nextIndex: number) {
+    currentBatchIndexRef.current = nextIndex;
+    forceRerender();
+  }
+
+  const prevAssetKeys = React.useRef(assetKeys);
+  if (prevAssetKeys.current !== assetKeys) {
+    currentBatchIndexRef.current = 0;
+    prevAssetKeys.current = assetKeys;
+  }
+
+  const numberOfBatches = React.useMemo(() => {
+    // Try fetching 50 at a time
+    return assetKeys.length / batchSize;
+  }, [assetKeys.length, batchSize]);
+
+  const currentBatch = React.useMemo(() => {
+    const startIndex = currentBatchIndex * batchSize;
+    return assetKeys.slice(startIndex, startIndex + batchSize);
+  }, [assetKeys, currentBatchIndex, batchSize]);
+
+  const allLiveDataByNodeRef = React.useRef<ReturnType<typeof buildLiveData>>({});
+
+  return {
+    currentBatch,
+    setBatchData: React.useCallback((liveDataByNode: typeof allLiveDataByNodeRef.current) => {
+      allLiveDataByNodeRef.current = Object.assign(
+        {},
+        allLiveDataByNodeRef.current,
+        liveDataByNode,
+      );
+      return allLiveDataByNodeRef.current;
+    }, []),
+    allLiveDataByNode: allLiveDataByNodeRef.current,
+    nextBatch: React.useCallback(() => {
+      setBatchIndex((currentBatchIndexRef.current + 1) % numberOfBatches);
+    }, []),
+    isLastBatch: numberOfBatches === currentBatchIndex + 1,
+  };
+};
```

Comment:
I think this makes sense and the code looks good, but my gut feeling is that all the complexity may not be worth it. If the asset group has 1000 assets in it and they load one at a time you'd have to sit on the page for... maybe 300s for them to all appear (assuming ~300ms per asset)?

It also seems like it may be best to fetch in larger batches (eg: 50 assets) because the backend does do a pretty good job of avoiding fan out in the SQL queries, etc., and I believe the `staleCauses` require it to load the data of related assets to return the data of a single asset.

It kinda sucks but I wonder if we could do what we do for `BackfillRow.tsx` (line 93 - if there are too many partitions, we just display nothing vs. trying to fetch the status of each item)

I guess this also impacts the main asset graph (where the user can see all thousand of these assets...) but that in that UI, we prompt them to narrow it down before rendering them all so they're more likely to ask for a subset.